### ISSUE TITLE
native plugins Promise Fails when using livereload

### ISSUE THREAD

**Issue by anaganisk** [driftyco/ionic-native#860]
I'm getting the following errors only when using livereload, without it the app works just fine.
I'm waiting for platform load, but i'm not sure what is causing this error
![1481382107](https://cloud.githubusercontent.com/assets/9448273/21074159/d9fe8ed4-bf17-11e6-9098-6fd85cd64f52.png)

**Comment by NiklasMerz** [comment]
I stumbled over this too

**Comment by ihadeed** [comment]
@anaganisk this is an issue with the CLI/App-Scripts. It's resolved in the master branches but not released yet. You can downgrade to older version of app scripts, run the app in non-livereload. Or do the following:

```
npm i --save-dev @ionic/app-scripts@nightly 
npm i -g ionic@test
```

and rename `main.dev.ts` to `main.ts`.

**Comment by anaganisk** [comment]
facing this error now after using the nightly release
```
0 info it worked if it ends with ok
1 verbose cli [ '/home/anaganisk/.nvm/versions/node/v6.9.2/bin/node',
1 verbose cli   '/home/anaganisk/.nvm/versions/node/v6.9.2/bin/npm',
1 verbose cli   'run',
1 verbose cli   'ionic:serve',
1 verbose cli   '--',
1 verbose cli   '--v2',
1 verbose cli   '--runLivereload',
1 verbose cli   '--isPlatformServe',
1 verbose cli   '--livereload',
1 verbose cli   '--address',
1 verbose cli   '192.168.0.100',
1 verbose cli   '--port',
1 verbose cli   '8000',
1 verbose cli   '--liveReloadPort',
1 verbose cli   '8000',
1 verbose cli   '--iscordovaserve',
1 verbose cli   '--nobrowser' ]
2 info using npm@3.10.9
3 info using node@v6.9.2
4 verbose run-script [ 'preionic:serve', 'ionic:serve', 'postionic:serve' ]
5 info lifecycle ionic-hello-world@~preionic:serve: ionic-hello-world@
6 silly lifecycle ionic-hello-world@~preionic:serve: no script for preionic:serve, continuing
7 info lifecycle ionic-hello-world@~ionic:serve: ionic-hello-world@
8 verbose lifecycle ionic-hello-world@~ionic:serve: unsafe-perm in lifecycle true
9 verbose lifecycle ionic-hello-world@~ionic:serve: PATH: /home/anaganisk/.nvm/versions/node/v6.9.2/lib/node_modules/npm/bin/node-gyp-bin:/home/anaganisk/Projects/bustrakr/mob/node_modules/.bin:/home/anaganisk/.nvm/versions/node/v6.9.2/bin:/home/anaganisk/bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/anaganisk/Android/Sdk/tools:/home/anaganisk/Android/Sdk/platform-tools:/home/anaganisk/Apps/esptool:/home/anaganisk/Apps/.utilities/phantomjs/bin:/home/anaganisk/Apps/logstash-5.0.0/bin:/home/anaganisk/Apps:/home/anaganisk/.nvm/versions/node/v6.9.2/bin
10 verbose lifecycle ionic-hello-world@~ionic:serve: CWD: /home/anaganisk/Projects/bustrakr/mob
11 silly lifecycle ionic-hello-world@~ionic:serve: Args: [ '-c',
11 silly lifecycle   'ionic-app-scripts serve "--v2" "--runLivereload" "--isPlatformServe" "--livereload" "--address" "192.168.0.100" "--port" "8000" "--liveReloadPort" "8000" "--iscordovaserve" "--nobrowser"' ]
12 silly lifecycle ionic-hello-world@~ionic:serve: Returned: code: 1  signal: null
13 info lifecycle ionic-hello-world@~ionic:serve: Failed to exec ionic:serve script
14 verbose stack Error: ionic-hello-world@ ionic:serve: `ionic-app-scripts serve "--v2" "--runLivereload" "--isPlatformServe" "--livereload" "--address" "192.168.0.100" "--port" "8000" "--liveReloadPort" "8000" "--iscordovaserve" "--nobrowser"`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/home/anaganisk/.nvm/versions/node/v6.9.2/lib/node_modules/npm/lib/utils/lifecycle.js:255:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (/home/anaganisk/.nvm/versions/node/v6.9.2/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:877:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid ionic-hello-world@
16 verbose cwd /home/anaganisk/Projects/bustrakr/mob
17 error Linux 4.4.0-53-generic
18 error argv "/home/anaganisk/.nvm/versions/node/v6.9.2/bin/node" "/home/anaganisk/.nvm/versions/node/v6.9.2/bin/npm" "run" "ionic:serve" "--" "--v2" "--runLivereload" "--isPlatformServe" "--livereload" "--address" "192.168.0.100" "--port" "8000" "--liveReloadPort" "8000" "--iscordovaserve" "--nobrowser"
19 error node v6.9.2
20 error npm  v3.10.9
21 error code ELIFECYCLE
22 error ionic-hello-world@ ionic:serve: `ionic-app-scripts serve "--v2" "--runLivereload" "--isPlatformServe" "--livereload" "--address" "192.168.0.100" "--port" "8000" "--liveReloadPort" "8000" "--iscordovaserve" "--nobrowser"`
22 error Exit status 1
23 error Failed at the ionic-hello-world@ ionic:serve script 'ionic-app-scripts serve "--v2" "--runLivereload" "--isPlatformServe" "--livereload" "--address" "192.168.0.100" "--port" "8000" "--liveReloadPort" "8000" "--iscordovaserve" "--nobrowser"'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the ionic-hello-world package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     ionic-app-scripts serve "--v2" "--runLivereload" "--isPlatformServe" "--livereload" "--address" "192.168.0.100" "--port" "8000" "--liveReloadPort" "8000" "--iscordovaserve" "--nobrowser"
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs ionic-hello-world
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls ionic-hello-world
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by ihadeed** [comment]
@anaganisk Can you try `@ionic/app-scripts@0.0.46-201612011929` .. This is the version I'm using and it works well. Not sure about the latest nightly that you got.

**Comment by anaganisk** [comment]
@ihadeed just tried 0.0.47 I don't think this issue is relevant anymore since app-scripts 0.0.47 fixes this issue, I think I need not use 0.0.46. Thank you!
